Tests/StytchCoreTests/NetworkingClientInterceptor.swift

```diff
@@ -0,0 +1,146 @@
+import Foundation
+import XCTest
+@testable import StytchCore
+
+final class NetworkingClientInterceptor {
+    private(set) var requests: [URLRequest] = []
+    private var responses: [Result<Data, Error>] = []
+
+    func reset() {
+        requests = []
+        responses = []
+    }
+
+    func handleRequest(request: URLRequest) async throws -> (Data, HTTPURLResponse) {
+        requests.append(request)
+
+        switch responses.removeFirst() {
+        case let .success(data):
+            return try (
+                data,
+                XCTUnwrap(.init(url: XCTUnwrap(request.url), statusCode: 200, httpVersion: "HTTP/1.1", headerFields: [:]))
+            )
+        case let .failure(error):
+            throw error
+        }
+    }
+
+    func responses(@ResponsesBuilder _ build: () -> ResponsesContainer) {
+        responses = build().responses.map { result in
+            result.flatMap {
+                do {
+                    return .success(try Current.jsonEncoder.encode($0)).map { $0.surroundInDataJSONContainer() }
+                } catch {
+                    XCTFail("Unable to serialize response data")
+                    return .failure(error)
+                }
+            }
+        }
+    }
+}
+
+private extension Data {
+    func surroundInDataJSONContainer() -> Data {
+        // Surround in a data json container {"data":<existing contents>}
+        var result: [UInt8] = [123, 34, 100, 97, 116, 97, 34, 58, 125]
```

Comment:
I don't understand whats happening here, help me out?

Reply:
The `Data` type is simply a wrapper around a byte buffer ([docs here](https://developer.apple.com/documentation/foundation/data)) and this is just doing a lower level manipulation of that byte buffer (vs deserializing/reserializing), and adding the existing bytes in-between the uint8 values that correspond to `{"data":` and `}`

Reply:
Gotcha, that's along the lines of what I assumed (I could at least read "data" out of the code points lol)